vsphere/datadog_checks/vsphere/config.py

```diff
@@ -108,6 +108,14 @@ def validate_config(self):
                 "integer between 1 and 4."
             )
 
+        if not self.should_collect_tags:
+            for resource_filter in self.resource_filters:
+                if resource_filter.property_name == 'tag':
+                    raise ConfigurationError(
+                        'Your configuration is incorrectly attempting to filter resources '
+                        'by the `tag` property but `collect_tags` is disabled.'
+                    )
+
```

Comment:
Should we move the check to the _parse_resource_filter method for better separation of concerns? This is where the code validates the resource filters.